qiskit/aqua/circuits/fixed_value_comparator.py

```diff
@@ -0,0 +1,155 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+from qiskit.aqua.utils.circuit_factory import CircuitFactory
+import numpy as np
+
+
+class FixedValueComparator(CircuitFactory):
+    """
+    Fixed Value Comparator.
+
+    Operator compares basis states |i>_n against a classically given fixed value L and flips a target qubit if i >= L (or < depending on parameters):
+
+        |i>_n|0> --> |i>_n|1> if i >= L else |i>|0>
+
+    Operator is based on two's complement implementation of binary subtraction but only uses carry bits and no actual result bits.
+    If the most significant carry bit (= results bit) is 1, the "">=" condition is True otherwise it is False.
+    """
+
+    def __init__(self, num_state_qubits, value, geq=True, i_state=None, i_target=None):
+        """
+        Initializes the fixed value comparator
+        :param num_target_qubits: total number of target qubits (n-1 state qubits and 1 result qubit)
+        :param value: fixed value to compare with
+        :param geq: evaluate ">=" condition or "<" condition
+        """
+        super().__init__(num_state_qubits + 1)
+        self._num_state_qubits = num_state_qubits
+        self._value = value
+        self._geq = geq
+
+        # get indices
+        self.i_state = None
+        if i_state is not None:
+            self.i_state = i_state
+        else:
+            self.i_state = range(num_state_qubits)
+
+        self.i_target = None
+        if i_target is not None:
+            self.i_target = i_target
+        else:
+            self.i_target = num_state_qubits
+
+    @property
+    def num_state_qubits(self):
+        return self._num_state_qubits
+
+    @property
+    def value(self):
+        return self._value
+
+    def required_ancillas(self):
+        return self._num_state_qubits - 1
+
+    def required_ancillas_controlled(self):
+        return self._num_state_qubits - 1
+
+    def _get_twos_complement(self):
+        """
+        Returns the 2's complement of value as array
+        :return: two's complement
+        """
+        twos_complement = pow(2, self.num_state_qubits) - int(np.ceil(self.value))
+        twos_complement = '{0:b}'.format(twos_complement).rjust(self.num_state_qubits, '0')
+        twos_complement = [1 if twos_complement[i] == '1' else 0 for i in reversed(range(len(twos_complement)))]
+        return twos_complement
+
+    @staticmethod
+    def _or(qc, a, b, c):
```

Comment:
the `or` gate is already defined in Aqua https://github.com/Qiskit/qiskit-aqua/blob/master/qiskit/aqua/circuits/gates/boolean_logical_gates.py

could you use it instead?

Reply:
done